### PR TITLE
Render mobile aside if needed

### PR DESCRIPTION
**Pull request by nickhsine**
### Dev Notes
`mobile-aside.js` and related svg assets  are copied from `twreporter-react` repo.

### CODE REVIEW

File: src/components/aside/mobile-aside.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import BackToTopicIcon from '../../assets/aside/article-back-to-topic-mobile.svg'
+import BackToTopIcon from '../../assets/aside/article-back-to-top-mobile.svg'
+import DynamicComponentsContext from '../../contexts/dynamic-components-context'
+import PropTypes from 'prop-types'
+import React from 'react'
+import mq from '../../utils/media-query'
+import soothScroll from 'smoothscroll'
```
Comment (YuCJ) [inline src/components/aside/mobile-aside.js]: typo `smoothScroll`

File: src/components/article-page.js [hunk 2]
```diff
@@ -200,6 +203,27 @@ export default class Article extends PureComponent {
...
+
+  componentDidMount() {
+    // detect sroll position
+    window.addEventListener('scroll', this.onScroll)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('scroll', this.onScroll)
+
+    this.mobileAsideRef = null
```
Comment (YuCJ) [inline src/components/article-page.js]: React will clear `this.mobileAsideRef.current` as `null` when the component unmount.
Ref. https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element

File: src/components/article-page.js [hunk 3]
```diff
@@ -241,6 +265,30 @@ export default class Article extends PureComponent {
...
+    if (this.mobileAsideRef) {
+      const mAside = this.mobileAsideRef
+      const currentTopY = window.scrollY
+
+      // Calculate scrolling distance to determine whether to display aside
+      const lastY = this.scrollPosition.y
+      const distance = currentTopY - lastY
+      if (distance > 30) {
+        this.scrollPosition.y = currentTopY
+        mAside.current.toShow = false
```
Comment (YuCJ) [inline src/components/article-page.js:281]: Will modifying the property of component instance directly make it re-render?

File: src/components/article-page.js [hunk 4]
```diff
@@ -297,8 +345,9 @@ export default class Article extends PureComponent {
             />
             <BodyBackground>
               <BodyBlock>
-                <AsideBlock>
-                  <Aside
+                <MobileAside toShow={true} ref={this.mobileAsideRef} />
```
Comment (YuCJ) [inline src/components/article-page.js]: `MobileAside` does not use the prop `toShow`

File: src/components/article-page.js (hunk 3 above)
Comment (nickhsine) [reply src/components/article-page.js:281]: `MobileAside.toShow` is a setter, which would execute `setState({toShow: toShow)` and re-render.

**Comment by nickhsine** [comment]
Clean commit log by rebasing.

**Pull request merged**